src/components/controls/basic/waveControl.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useContext, useState, useEffect } from 'react';
+import { getAllWaveTypes } from "../../../data/waveforms";
+import styled from "styled-components";
+import StyledLabel from "../../styled/controlLabels";
+import StyledSelect from "../../styled/controlSelect";
+import IKeyboardContextSignature from "../../../interfaces/IKeyboardContextSignature";
+
+import { KeyboardContext, UpdateKeyboardContext } from "../../../hooks/keyboardContext";
+
+const StyledWaveControl = styled.div`
+    display:flex;
+    flex-direction:column;
+    align-items:center;
+    justify-content:center;
+`
+
+function WaveControl(){
+    const state : IKeyboardContextSignature = useContext(KeyboardContext);
+    const setState = useContext(UpdateKeyboardContext);
+    const localStorageKey = "waveform";
+    const waveTypeOptions = getAllWaveTypes();
+  
+    const { audioContextWrapper }: IKeyboardContextSignature = useContext(KeyboardContext);
+    let initialWaveform = localStorage.getItem(localStorageKey) ?? waveTypeOptions[0];
+
+    const [wave, setWave] = useState(initialWaveform);
```

Comment:
Can you explain why we need both `useState` and `useContext`? I thought the state was included when we got the context?

Reply:
useState keeps track of the state within the component. UseContext is used to have access to the global audioContextWrapper